csharp/src/Apache.Arrow.Adbc/AdbcConnection11.cs

```diff
@@ -0,0 +1,390 @@
+        /// Rollback the pending transaction.
+        /// </summary>
+        public virtual Task RollbackAsync(CancellationToken cancellationToken = default)
+        {
+            throw AdbcException.NotImplemented("Connection does not support transactions");
+        }
+
+        /// <summary>
+        /// Gets the names of statistics specific to this driver.
+        /// </summary>
+        /// <returns></returns>
+        public virtual IArrowArrayStream GetStatisticsNames()
+        {
+            throw AdbcException.NotImplemented("Connection does not support statistics");
+        }
+
+        /// <summary>
+        /// Gets the names of statistics specific to this driver.
+        /// </summary>
+        /// <returns></returns>
+        public virtual ValueTask<IArrowArrayStream> GetStatisticsNamesAsync(CancellationToken cancellationToken = default)
+        {
+            return new ValueTask<IArrowArrayStream>(GetStatisticsNames());
+        }
+
+        /// <summary>
+        /// Gets statistics about the data distribution of table(s)
+        /// </summary>
+        /// <param name="catalogPattern">The catalog or null. May be a search pattern.</param>
+        /// <param name="schemaPattern">The schema or null. May be a search pattern.</param>
+        /// <param name="tableName">The table name or null. May be a search pattern.</param>
+        /// <param name="approximate">If false, consumer desires exact statistics regardless of cost</param>
+        /// <returns>A result describing the statistics for the table(s)</returns>
+        public virtual IArrowArrayStream GetStatistics(string? catalogPattern, string? schemaPattern, string tableNamePattern, bool approximate)
+        {
+            return Task.Run(() => GetStatisticsAsync(catalogPattern, schemaPattern, tableNamePattern, approximate)).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Gets statistics about the data distribution of table(s)
+        /// </summary>
+        /// <param name="catalogPattern">The catalog or null. May be a search pattern.</param>
+        /// <param name="schemaPattern">The schema or null. May be a search pattern.</param>
+        /// <param name="tableName">The table name or null. May be a search pattern.</param>
+        /// <param name="approximate">If false, consumer desires exact statistics regardless of cost</param>
+        /// <returns>A result describing the statistics for the table(s)</returns>
+        public virtual Task<IArrowArrayStream> GetStatisticsAsync(string? catalogPattern, string? schemaPattern, string tableNamePattern, bool approximate)
+        {
+            throw AdbcException.NotImplemented("Connection does not support statistics");
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+
+        protected virtual void Dispose(bool disposing)
+        {
+        }
+
+        public ValueTask DisposeAsync()
```

Comment:
does this need an equivalent #if NET5_0_OR_GREATER directive?

Reply:
It's not strictly required because you can define whatever function you want; only the interface it implements is missing in older versions of .NET. I guess my design principle here is to #ifdef only when necessary; but does that give a misleading impression in this case?